lingua_franca/internal.py

```diff
@@ -0,0 +1,503 @@
+import os.path
+from functools import wraps
+from importlib import import_module
+from inspect import signature
+from warnings import warn
+
+_SUPPORTED_LANGUAGES = ("cs", "da", "de", "en", "es", "fr", "hu",
+                        "it", "nl", "pt", "sv")
+
+_SUPPORTED_FULL_LOCALIZATIONS = ("cs-cz", "da-dk", "de-de", "en-au", "en-us",
+                                 "es-es", "fr-fr", "hu-hu", "it-it", "nl-nl",
+                                 "pt-pt", "ru-ru", "sv-se", "tr-tr")
+
+_DEFAULT_FULL_LANG_CODES = {'cs': 'cs-cz',
+                            'da': 'da-dk',
+                            'de': 'de-de',
+                            'en': 'en-us',
+                            'es': 'es-es',
+                            'fr': 'fr-fr',
+                            'hu': 'hu-hu',
+                            'it': 'it-it',
+                            'nl': 'nl-nl',
+                            'pt': 'pt-pt',
+                            'ru': 'ru-ru',
+                            'sv': 'sv-se',
+                            'tr': 'tr-tr'}
+
+__default_lang = None
+__active_lang_code = None
+__loaded_langs = []
+
+_localized_functions = {}
+
+
+class UnsupportedLanguageError(NotImplementedError):
+    pass
+
+
+class FunctionNotLocalizedError(NotImplementedError):
+    pass
+
+
+def raise_unsupported_language(language):
+    """
+    Raise an error when a language is unsupported
+
+    Arguments:
+        language: str
+            The language that was supplied.
+    """
+    supported = ' '.join(_SUPPORTED_LANGUAGES)
+    raise UnsupportedLanguageError("\nLanguage '{language}' is not yet "
+                                   "supported by Lingua Franca. "
+                                   "Supported language codes "
+                                   "include the following:\n{supported}"
+                                   .format(language=language, supported=supported))
+
+
+def get_supported_langs():
+    return _SUPPORTED_LANGUAGES
+
+
+def get_active_langs():
+    """ Get the list of currently-loaded language codes
+
+    Returns:
+        list(str)
+    """
+    return __loaded_langs
+
+
+def set_active_langs(langs=_SUPPORTED_LANGUAGES, override_default=True):
+    """ Set the list of languages to load.
```

Comment:
This doesn't need to be public, if you don't think it's appropriate. I decided to leave it open for convenience, so that Mycroft won't have to do

```python3
for lang in config.languages:
  lingua_franca.load_language(lang)
lingua_franca.set_default...
```

which isn't hard, it just doesn't have a snappy and obvious name like `load_languages()`. This function also shoves the first element into the default slot, which might *not* be obvious.

Reply:
Yeah I liked the first element becoming default, that shouldn't be affected by making this private should it?